js/views/QuickAdd.svelte

```diff
@@ -100,16 +104,39 @@
 			bind:this={noteInput} />
 	</label>
 	<label for="quick-add-time">
-		{@html translate('timemanager', 'Duration (in hrs.) & Date')}
+		{@html translate('timemanager', 'Duration (in hrs.), Start time, End time & Date')}
 		<span class="double">
 			<input
 				id="quick-add-time"
 				type="number"
 				name="duration"
 				step="0.01"
+				min="0"
+				max="23.99"
 				placeholder=""
 				class="duration-input"
-				bind:value={duration} />
+				bind:value={duration}
+				on:input={() => endTime = Helpers.calculateEndTime(startTime, duration)} />
+			<input
```

Comment:
One issue we have here is that this label now includes 3 inputs. Ideally, we'd have one label per input, but I can see how this would be difficult design-wise. See my comment about adding switch to toggle between time/hours mode. Maybe that implementation would make it easier to have a separate label for each input 🤓

Reply:
No changes done here yet.